pkg/registries/set_impl.go

```diff
@@ -49,6 +50,27 @@ func (e *setImpl) GetAll() []types.ImageRegistry {
 	return e.getSortedRegistriesNoLock()
 }
 
+// GetAllUnique returns the unique set of integrations that are active.
+// Unique integrations are dependent on the integrations URL and auth information.
+func (e *setImpl) GetAllUnique() []types.ImageRegistry {
```

Comment:
When would someone want to use `GetAll` instead of this?

Reply:
Can you give me an example scenario what was previously problematic that this fixes? Do you think that scenario was expected to have the effect that it did? As in, was this new function meant to be the original functionality?

Reply:
So you would use `GetAll` in places where you want to show all current integrations, such as the UI and other places that display those.

As to what this fixes: often times, when customers have many secured clusters, they also share credentials for specific registries across those clusters. This then leads to multiple autogenerated image integrations being created and pushed to Central with the same config (i.e. endpoint, username, password). During reprocessing, when scanning images, we will iterate through all of those integrations and try them out one by one. This leads to increased traffic to registries which ideally we shouldn't do, since if the credentials for the _same_ registry URL haven't worked once, chances are they don't work the other N times the integration is duplicated.

I don't think this was meant to be the original function, initially we didn't expect this scale of secured clusters we have been seeing nowadays.